src/transact/update-pending-tx.js

```diff
@@ -1,45 +1,43 @@
 "use strict";
 
-var clone = require("clone");
+var assign = require("lodash.assign");
 var eth = require("../wrappers/eth");
 var updateMinedTx = require("../transact/update-mined-tx");
 var transact = require("../transact/transact");
 var isFunction = require("../utils/is-function");
+var logError = require("../utils/log-error");
 var errors = require("../errors/codes");
+var RPCError = require("../errors/rpc-error");
 var constants = require("../constants");
 
 function updatePendingTx(txHash) {
   return function (dispatch, getState) {
-    var currentBlock;
-    dispatch(eth.getTransactionByHash(txHash, function (onChainTx) {
-      var e, storedTransaction;
+    var storedTransaction = getState().transactions[txHash];
+    var onFailed = isFunction(storedTransaction.onFailed) ? storedTransaction.onFailed : logError;
+    dispatch(eth.getTransactionByHash(txHash, function (err, onChainTx) {
+      if (err) return onFailed(err);
       dispatch({
         type: "UPDATE_TRANSACTION",
         hash: txHash,
-        data: { tx: onChainTx || {} }
+        data: { tx: onChainTx || {} },
       });
 
       // if transaction is null, then it was dropped from the txpool
       if (onChainTx === null) {
```

Comment:
I believe this check is specifically for `null` (rather than `null` or `undefined`) -- it's checking specifically for the odd case where the transaction is dropped from the transaction pool, which results in `null` when you try and lookup the transaction.  (NB: should double-check whether this is true in Parity; I know it is true in geth.)